File: exercism/csharp/hexadecimal/Hexadecimal.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Hexadecimal
{
    static string Hex = "0123456789abcdef";

    public static int ToDecimal(string digits)
    {
        IEnumerable<int> vals =
            digits
            .Reverse()
            .Select((d, i) => (int)Math.Pow(16, i) * Hex.IndexOf(d));
        if (vals.Where(n => n == -1).Any()) return 0;
        return vals.Sum();
    }
}

File: exercism/cpp/hexadecimal/hexadecimal.cpp
#include "hexadecimal.h"

uint64_t hexadecimal::convert(std::string digits) {
  std::string hex = "0123456789abcdef";
  int len = digits.length();
  int sum = 0;
  for (int i = 0; i < len; i++) {
    int val = hex.find(digits[i], 0);
    if (val < 0 || val > 15)
      return 0;
    sum <<= 4;
    sum += val;
  }
  return sum;
}
